specs/core/0_beacon-chain.md

```diff
@@ -805,17 +816,24 @@ def get_randao_mix(state: BeaconState,
     return state.latest_randao_mixes[epoch % LATEST_RANDAO_MIXES_LENGTH]
 ```
 
-### `get_active_index_root`
+### `get_active_registry_root`
 
 ```python
-def get_active_index_root(state: BeaconState,
-                          epoch: Epoch) -> Bytes32:
+def get_active_registry_root(state: BeaconState) -> Bytes32:
     """
-    Return the index root at a recent ``epoch``.
-    ``epoch`` expected to be between
-    (current_epoch - LATEST_ACTIVE_INDEX_ROOTS_LENGTH + ACTIVATION_EXIT_DELAY, current_epoch + ACTIVATION_EXIT_DELAY].
+    Return the active registry root for the current epoch.
     """
-    return state.latest_active_index_roots[epoch % LATEST_ACTIVE_INDEX_ROOTS_LENGTH]
+    epoch = get_current_epoch(state)
+    active_registry = ActiveRegistry()
+    for committee_number in range(get_epoch_committee_count(state, epoch)):
+        shard = (get_start_shard(state, epoch) + committee_number) % SHARD_COUNT
+        for index in get_crosslink_committee(state, epoch, shard):
+            validator = validator_registry[index]
+            active_registry[shard].pubkeys.append(validator.pubkey)
```

Comment:
You're sorting validators by their _crosslink_ committee shard. This isn't going to be particularly useful for light clients _if_ we don't merge the two committees.

Reply:
That's a good point. Maybe we want to wait until phase 1 to put this light-client infrastructure in place.

Reply:
The `active_index_roots` in combination with the historical randaomixes being in state was generic enough to serve proofs about the historic crosslink and persistent committees.
If we go this new path, we should either wait or intend to put both committees in the registry root.